### PR TITLE
Feature: (ISA-135) Remove category specific logic (catalogue panels)

### PR DESCRIPTION
**Pull request by JoshGx2000**
https://jira.its.utas.edu.au/browse/ISA-135

Adds category info to the DB, and uses new category display name field for the catalogue buttons and panels text.

### CODE REVIEW

File: backend/catalogue/viewsets.py [hunk 1]
```diff
@@ -3,8 +3,8 @@
 # Released under the Affero General Public Licence (AGPL) v3.  See LICENSE file for details.
 from django.db.models import Value
 from django.db.models.functions import Coalesce
-from catalogue.models import Organisation, HabitatDescriptor, BaseLayer, Layer, LayerGroup, LayerGroupPriority, DataClassification, BaseLayerGroup, SaveState
```
Comment (markhepburn) [inline backend/catalogue/viewsets.py:6]: These are getting quite long -- let's clean up while we're at it ("import ..as models", then refer as models.BaseLayer, etc)

File: backend/webapp/urls.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 from django.urls import include, re_path
 from rest_framework.routers import DefaultRouter
 
-from catalogue.viewsets import ClassificationViewset, DescriptorViewset, BaseLayerViewset, LayerViewset, GroupViewset, GroupPriorityViewset, OrganisationViewset, BaseLayerGroupViewset
```
Comment (markhepburn) [inline backend/webapp/urls.py:9]: Similar qualified import (import-as) could be used here.  (I know this is my fault that I'm asking you to clean up!)

File: frontend/src/cljs/imas_seamap/map/subs.cljs [hunk 3]
```diff
@@ -65,6 +65,13 @@
 (defn map-base-layers [{:keys [map]} _]
   (select-keys map [:grouped-base-layers :active-base-layer]))
 
+(defn categories-map [db _]
+  (let [categories (get-in db [:map :categories])]
+    (reduce
+     (fn [categories {:keys [name] :as category}]
```
Comment (markhepburn) [inline frontend/src/cljs/imas_seamap/map/subs.cljs:71]: Second time in this PR I've seen this pattern (I always zoom in on uses of reduce, it's a prime candidate for "actually there's an existing library fn that does this").  How about extracting a util to index a collection by a key?  (If there isn't one already).  Have a bit of a play; one option is just to extract the categories case, another might be to pull out the map-key function, eg (into-map :name categories) or something like that (not claiming that's the best name)

File: backend/catalogue/viewsets.py (hunk 1 above)
Comment (JoshGx2000) [reply backend/catalogue/viewsets.py:6]: Good idea

File: frontend/src/cljs/imas_seamap/map/subs.cljs (hunk 3 above)
Comment (JoshGx2000) [reply frontend/src/cljs/imas_seamap/map/subs.cljs]: I've extracted the function, as after some searching I've found examples of similar problems, which are not solved by a library function (e.g. https://stackoverflow.com/questions/47082635/map-from-list-of-maps).

File: backend/catalogue/viewsets.py (hunk 1 above)
Comment (JoshGx2000) [reply backend/catalogue/viewsets.py:6]: Imports cleaned-up

File: backend/webapp/urls.py (hunk 2 above)
Comment (JoshGx2000) [reply backend/webapp/urls.py:9]: Imports cleaned-up

**Pull request merged**